lab_3_ann_retriever/main.py

```diff
@@ -105,6 +106,14 @@ def tokenize(self, text: str) -> list[str] | None:
         In case of corrupt input arguments, None is returned.
         """
 
+        if any((not isinstance(text, str), not text)):
+            return None
+
+        for sym in text:
+            if all((not sym.isalpha(), sym != ' ')):
```

Comment:
Actually, I do not understand this line too

Reply:
it's supposed to be a condition which checks if the sym in text is a non-alphabetic symbol and not a space, because later it would be replaced with the same space. The same algorithm was used in the previous labs.

The Python all() function returns true if all the elements of a given iterable (List, Dictionary, Tuple, set, etc.) are True otherwise it returns False.

Should I replace:
if all((not sym.isalpha(), sym != ' '))

With:
if not sym.isalpha() and sym != ' '   ?